spec/lib/submission_builder/ty2024/states/nj/nj_return_xml_spec.rb

```diff
@@ -24,6 +24,16 @@
         expect(xml.document.at('ReturnDataState FormNJ1040 Header')).to be_an_instance_of Nokogiri::XML::Element
       end
 
+      context "when there is a refund with banking info" do
+        let(:intake) { create(:state_file_nj_refund_intake) }
+  
+        it "generates FinancialTransaction xml without RefundAmt" do
+          allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:refund_or_owed_amount).and_return 500
+          xml = Nokogiri::XML::Document.parse(described_class.build(submission).document.to_xml)
+          expect(xml.at("FinancialTransaction")).to be_present
```

Comment:
[Pebble] Can we also assert on that the children of this element exist and are correct?

RefundDirectDeposit exists
RoutingTransitNumber matches the factory value
BankAccountNumber matches the factory value
Amount matches the mocked value on line 31
Savings is checked based on factory
NotIATTransaction is set correctly (not sure how that logic works)

Reply:
That's covered in financial transaction spec

Reply:
Makes sense. Feel free to disregard and resolve then.